mailbox/store/src/test/java/org/apache/james/mailbox/store/extractor/JsoupTextExtractorTest.java

```diff
@@ -57,15 +58,23 @@ void extractedTextFromHtmlShouldNotContainTheContentOfTitleTag() throws Exceptio
     void extractContentShouldHandlePlainText() throws Exception {
         InputStream inputStream = new ByteArrayInputStream("myText".getBytes(StandardCharsets.UTF_8));
 
-        assertThat(textExtractor.extractContent(inputStream, "text/plain").getTextualContent())
+        assertThat(textExtractor.extractContent(inputStream, ContentType.of("text/plain")).getTextualContent())
+                .contains("myText");
+    }
+
+    @Test
+    void extractContentShouldHandlePlainTextWithCharset() throws Exception {
+        InputStream inputStream = new ByteArrayInputStream("myText".getBytes(StandardCharsets.UTF_8));
+
+        assertThat(textExtractor.extractContent(inputStream, ContentType.of("text/plain; charset=utf-8")).getTextualContent())
                 .contains("myText");
     }
 
     @Test
     void extractContentShouldHandleArbitraryTextMediaType() throws Exception {
         InputStream inputStream = new ByteArrayInputStream("myText".getBytes(StandardCharsets.UTF_8));
 
-        assertThat(textExtractor.extractContent(inputStream, "text/arbitrary").getTextualContent())
+        assertThat(textExtractor.extractContent(inputStream, ContentType.of("text/arbitrary")).getTextualContent())
                 .isEmpty();
     }
```

Comment:
could you add some tests with different charsets?

Reply:
Done :-)